darts/models/theta.py

```
@@ -115,14 +123,228 @@ def predict(self, n: int) -> 'TimeSeries':
+
+    def predict(self, n: int) -> 'TimeSeries':
+        super().predict(n)
+
+        # Forecast of the SES part.
+        forecast = self.model.forecast(n)
+
+        # Forecast of the Linear Regression part.
+        drift = self.drift(np.arange(self.length, self.length + n))
+        if self.trend_mode is Trend.EXPONENTIAL:
+            drift = np.exp(drift)
+
+        if self.model_mode is Model.ADDITIVE:
+            forecast = self.wses * forecast + self.wdrift * drift
+        else:
+            forecast = forecast**self.wses * drift**self.wdrift
+
+        # Re-apply the seasonal trend of the TimeSeries
+        if self.is_seasonal:
+
+            replicated_seasonality = np.tile(self.seasonality.pd_series()[-self.season_period:],
+                                             math.ceil(n / self.season_period))[:n]
+            if self.season_mode is Season.MULTIPLICATIVE:
+                forecast *= replicated_seasonality
+            else:
+                forecast += replicated_seasonality
+
+        forecast *= self.mean
+
+        return self._build_forecast_series(forecast)
+
+    @staticmethod
+    def select_best_model(ts: TimeSeries, thetas: List[int] = None, m: Optional[int] = None):
+        """
+        Performs a grid search over all hyper parameters to select the best model.
+
+        Parameters
+        ----------
+        ts
+            The TimeSeries on which the model will be tested.
+        thetas
+            A list of thetas to loop over.
+        m
+            Optionally, the season used to decompose the time series.
+        Returns
+        -------
+        theta
+            The best performing model on the time series.
+        """
+        # Only import if needed
+        from ..backtesting.backtesting import backtest_gridsearch
+        from sklearn.metrics import mean_absolute_error as mae
+        if thetas is None:
+            thetas = [1, 2, 3]
+        elif isinstance(thetas, int):
```

I would remove this case, that's abusing the types, better let it crash IMO.